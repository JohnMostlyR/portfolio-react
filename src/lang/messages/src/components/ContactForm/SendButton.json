[
  {
    "id": "send_button.idle",
    "description": "Send button value in normal, idle, state.",
    "defaultMessage": "Send"
  },
  {
    "id": "send_button.sending",
    "description": "Send button value when data is send.",
    "defaultMessage": "Sending..."
  },
  {
    "id": "send_button.success",
    "description": "Send button value when data is successfully sent.",
    "defaultMessage": "Sent! I'll respond as soon as possible."
  },
  {
    "id": "send_button.error",
    "description": "Send button value when an error occurred.",
    "defaultMessage": "Failed. Please try again"
  }
]