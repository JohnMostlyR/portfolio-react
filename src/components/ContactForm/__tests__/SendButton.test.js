import React from 'react';
import {mount} from 'enzyme';
import toJson from 'enzyme-to-json';

import SendButton from '../SendButton';

describe('SendButton', () => {
  it('should render and match the snapshot', () => {
    const wrapper = mount(<SendButton/>);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });

  it('should have the correct default props', () => {
    const wrapper = mount(<SendButton/>);
    expect(wrapper.prop('disabled')).toBe(true);
    expect(wrapper.prop('buttonState')).toBe('idle');
  });

  it('should respond when set to `enabled`', () => {
    const wrapper = mount(<SendButton disabled={false}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });

  describe('when the `state` changes', () => {
    let wrapper;

    afterEach(() => {
      wrapper.unmount();
    });

    it('should change its appearance when state is `sending`', () => {
      wrapper = mount(<SendButton disabled={false} buttonState={'sending'} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should change its appearance when state is `success`', () => {
      wrapper = mount(<SendButton disabled={false} buttonState={'success'} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should change its appearance when state is `error`', () => {
      wrapper = mount(<SendButton disabled={false} buttonState={'error'} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('TODO: TEST THIS: should only be allowed to be given a select set of states', () => {
      // TODO: should test this but at the moment a bit lost as to how.
    });
  });
});