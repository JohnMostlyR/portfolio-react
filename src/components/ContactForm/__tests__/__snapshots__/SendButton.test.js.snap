// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SendButton should render and match the snapshot 1`] = `
<SendButton
  buttonState="idle"
  disabled={true}
>
  <styled.button
    disabled={true}
  >
    <button
      className="sc-bdVaJa iFebLj"
      disabled={true}
    >
      <span>
        <Styled(FontAwesome)
          name="paper-plane"
        >
          <FontAwesome
            className="sc-bwzfXH hueGiT"
            name="paper-plane"
          >
            <span
              aria-hidden={true}
              className="fa fa-paper-plane sc-bwzfXH hueGiT"
            />
          </FontAwesome>
        </Styled(FontAwesome)>
        <span>
          <FormattedMessage
            defaultMessage="Send"
            id="send_button.idle"
          >
            <IntlProvider
              defaultLocale="en"
              locale="en"
            >
              <FormattedMessage
                defaultMessage="Send"
                id="send_button.idle"
                values={Object {}}
              >
                <span>
                  Send
                </span>
              </FormattedMessage>
            </IntlProvider>
          </FormattedMessage>
        </span>
      </span>
    </button>
  </styled.button>
</SendButton>
`;

exports[`SendButton should respond when set to \`enabled\` 1`] = `
<SendButton
  buttonState="idle"
  disabled={false}
>
  <styled.button
    disabled={false}
  >
    <button
      className="sc-bdVaJa iFebLj"
      disabled={false}
    >
      <span>
        <Styled(FontAwesome)
          name="paper-plane"
        >
          <FontAwesome
            className="sc-bwzfXH hueGiT"
            name="paper-plane"
          >
            <span
              aria-hidden={true}
              className="fa fa-paper-plane sc-bwzfXH hueGiT"
            />
          </FontAwesome>
        </Styled(FontAwesome)>
        <span>
          <FormattedMessage
            defaultMessage="Send"
            id="send_button.idle"
          >
            <IntlProvider
              defaultLocale="en"
              locale="en"
            >
              <FormattedMessage
                defaultMessage="Send"
                id="send_button.idle"
                values={Object {}}
              >
                <span>
                  Send
                </span>
              </FormattedMessage>
            </IntlProvider>
          </FormattedMessage>
        </span>
      </span>
    </button>
  </styled.button>
</SendButton>
`;

exports[`SendButton when the \`state\` changes should change its appearance when state is \`error\` 1`] = `
<SendButton
  buttonState="error"
  disabled={false}
>
  <styled.button
    disabled={false}
  >
    <button
      className="sc-bdVaJa iFebLj"
      disabled={false}
    >
      <span>
        <Styled(FontAwesome)
          name="paper-plane"
        >
          <FontAwesome
            className="sc-bwzfXH hueGiT"
            name="paper-plane"
          >
            <span
              aria-hidden={true}
              className="fa fa-paper-plane sc-bwzfXH hueGiT"
            />
          </FontAwesome>
        </Styled(FontAwesome)>
        <span>
          <FormattedMessage
            defaultMessage="Failed. Please try again"
            id="send_button.error"
          >
            <IntlProvider
              defaultLocale="en"
              locale="en"
            >
              <FormattedMessage
                defaultMessage="Failed. Please try again"
                id="send_button.error"
                values={Object {}}
              >
                <span>
                  Failed. Please try again
                </span>
              </FormattedMessage>
            </IntlProvider>
          </FormattedMessage>
        </span>
      </span>
    </button>
  </styled.button>
</SendButton>
`;

exports[`SendButton when the \`state\` changes should change its appearance when state is \`sending\` 1`] = `
<SendButton
  buttonState="sending"
  disabled={false}
>
  <styled.button
    disabled={false}
  >
    <button
      className="sc-bdVaJa iFebLj"
      disabled={false}
    >
      <span>
        <Styled(FontAwesome)
          name="paper-plane"
        >
          <FontAwesome
            className="sc-bwzfXH hueGiT"
            name="paper-plane"
          >
            <span
              aria-hidden={true}
              className="fa fa-paper-plane sc-bwzfXH hueGiT"
            />
          </FontAwesome>
        </Styled(FontAwesome)>
        <span>
          <FormattedMessage
            defaultMessage="Sending..."
            id="send_button.sending"
          >
            <IntlProvider
              defaultLocale="en"
              locale="en"
            >
              <FormattedMessage
                defaultMessage="Sending..."
                id="send_button.sending"
                values={Object {}}
              >
                <span>
                  Sending...
                </span>
              </FormattedMessage>
            </IntlProvider>
          </FormattedMessage>
        </span>
      </span>
    </button>
  </styled.button>
</SendButton>
`;

exports[`SendButton when the \`state\` changes should change its appearance when state is \`success\` 1`] = `
<SendButton
  buttonState="success"
  disabled={false}
>
  <styled.button
    disabled={false}
  >
    <button
      className="sc-bdVaJa iFebLj"
      disabled={false}
    >
      <span>
        <Styled(FontAwesome)
          name="paper-plane"
        >
          <FontAwesome
            className="sc-bwzfXH hueGiT"
            name="paper-plane"
          >
            <span
              aria-hidden={true}
              className="fa fa-paper-plane sc-bwzfXH hueGiT"
            />
          </FontAwesome>
        </Styled(FontAwesome)>
        <span>
          <FormattedMessage
            defaultMessage="Sent! I'll respond as soon as possible."
            id="send_button.success"
          >
            <IntlProvider
              defaultLocale="en"
              locale="en"
            >
              <FormattedMessage
                defaultMessage="Sent! I'll respond as soon as possible."
                id="send_button.success"
                values={Object {}}
              >
                <span>
                  Sent! I'll respond as soon as possible.
                </span>
              </FormattedMessage>
            </IntlProvider>
          </FormattedMessage>
        </span>
      </span>
    </button>
  </styled.button>
</SendButton>
`;
